xarray/core/daskmanager.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Any, Callable
+
+import numpy as np
+
+from xarray.core.duck_array_ops import dask_available
+from xarray.core.indexing import ImplicitToExplicitIndexingAdapter
+from xarray.core.parallelcompat import ChunkManagerEntrypoint, T_ChunkedArray
+from xarray.core.pycompat import is_duck_dask_array
+
+if TYPE_CHECKING:
+    from xarray.core.types import DaskArray, T_Chunks, T_NormalizedChunks
+
+
+class DaskManager(ChunkManagerEntrypoint["DaskArray"]):
+    array_cls: type[DaskArray]
```

Comment:
does this work?
```suggestion
    array_cls: type[DaskArray] = array_type("dask")
```

Reply:
With that change the code works, but we eagerly import dask:

```python
FAILED xarray/tests/test_plugins.py::test_lazy_import - AssertionError: {'dask.array', 'scipy'} have been imported but should be lazy
```

It's an interesting suggestion though because (aside from the eager import) it means I can get rid of the `__init__` method on `ChunkManager`, showing that the chunkmanager's role really is just like a namespace.

Reply:
I also had to do a funky cast in order to satisfy mypy for the case that dask is not available

```python
array_cls = cast(type["DaskArray"], array_type("dask"))  # cast avoids mypy complaint when dask is not available
```

Reply:
> With that change the code works, but then we eagerly import dask

If the `daskmanager` were defined in another package, e.g. `dask-xarray` or `dask` itself, the import would become lazy. If we were adding dask integration to xarray for the first time today, this is the setup I would be suggesting.